config/site/site.config.js

```diff
@@ -0,0 +1,122 @@
+    customPatterns: [
+      // {
+      //   pattern: 'dir/lib/terra-dev-site/**/*.<type>.{jsx,js})',
+      //   entryPoint: path.join(process.cwd(),'lib', 'terra-dev-site'),
+      // },
+    ],
+  },
+
+  /* Hot reloading section. On by default for dev builds. This will automatically alias the dist dir as the source dir if you use the bundled webpack config. */
+  hotReloading: {
+    enabled: true,
+    source: 'src',
+    dist: 'lib',
+  },
+
+  /* Additional alias to add to webpack, see webpacks doc for more info on aliases. */
+  webpackAliases: {
+    // [path]: <path location>
+  },
+
+  /* The mono repo package directory. If this is found aliases will be setup for the mono repo packages by default. */
+  monoRepoPackageDir: path.resolve(process.cwd(), 'packages'),
+
+  /* The actual parced package.json file. May need to change this if you have a non-standard package.json path. */
+  npmPackage,
+
+  /* Side effect theme imports */
+  themeImports: [
+  //   './theme';
+  ],
+
+  /* The image to display as page placeholder when a component does not render. */
+  placeholderSrc: 'https://github.com/cerner/terra-dev-site/raw/master/terra.png',
+
+  /* The README content to display on the home page. */
+  readMeContent: path.resolve(process.cwd(), 'README.md'),
+
+  appConfig: {
+    /* The logo the site header should display. */
+    logoSrc: 'https://github.com/cerner/terra-dev-site/raw/master/terra.png', // maps to appLogoSrc
+
+    /* The title the site header should display. */
+    title: startCase(npmPackage.name), // maps to appTitle
+
+    /** The themes to supply the ThemeProvider which allows the site to switch
+      * between themes. Providing multiple enables the theme utility to display
+      * in the header's toolbar.
+      */
+    themes: {
+      'Default Theme': '',
+    },
+
+    /** The default theme of the site. Note, this value should be a key that was
+     * supplied to the themes object. The open-sourced theme is the default theme.
+     */
+    defaultTheme: 'Default Theme',
+
+    /** The locales to supply Base with, which allows the site to switch
+      * between locales. English is the default language.
+      */
+    locales: i18nSupportedLocales,
+
+    /* The default locale of the site. 'en' is the default theme. */
+    defaultLocale: 'en',
+
+    /** Indicates if the site supports bidirectionality. If enabled, the directionality
+      * utility will display in the toolbar.
+      */
+    bidirectional: true,
+
+
+    /* The default direction of the site. 'ltr' is the default direction. */
+    defaultDirection: 'ltr',
+
+    /* Configuration relating to the extensions section of the site. Not rendered if extensions are supplied by navigation config */
+    // extensions: {
+    // //    The url to link to github. If this is supplied a github extention will be display to link to the supplied url
+    //   gitHubUrl: 'https://github.com/cerner',
+    // },
+
+    ...(npmPackage.repository.url) && {
```

Comment:
curious why the `...`

Reply:
this is some magic @StephenEsser taught me. This will only include the key if the expression evaluates to true

Reply:
![mind_blown](https://user-images.githubusercontent.com/14099737/40237424-0858747a-5a76-11e8-9858-79c9151d912b.gif)

Reply:
![image](https://media2.giphy.com/media/12NUbkX6p4xOO4/giphy.gif)